akka-diagnostics/src/test/java/akka/diagnostics/ConfigCheckerSpec.scala

```diff
@@ -0,0 +1,1087 @@
+
+      warnings should be(Vector.empty)
+    }
+
+    "not warn about the dynamic hostnames when artery is used" in {
+      val config1 = ConfigFactory
+        .parseString("""
+       akka {
+         actor {
+           provider = remote
+         }
+         remote {
+           artery {
+             enabled = on
+             canonical.hostname = "<getHostAddress>"
+             canonical.port = 25252
+             log-aeron-counters = on
+           }
+         }
+       }
+      """)
+        .withFallback(reference)
+
+      val checker = new ConfigChecker(extSys, config1, reference)
+      val warnings = checker.check().warnings
+
+      printDocWarnings(warnings)
+      assertCheckerKey(warnings, "hostname")
+      assertPath(warnings, "akka.remote.artery.canonical.hostname")
+
+      val config2 =
+        ConfigFactory.parseString("""akka.remote.artery.canonical.hostname = "<getHostName>" """).withFallback(config1)
+
+      val checker2 = new ConfigChecker(extSys, config2, reference)
+      val warnings2 = checker2.check().warnings
+
+      printDocWarnings(warnings2)
+      assertCheckerKey(warnings2, "hostname")
+      assertPath(warnings2, "akka.remote.artery.canonical.hostname")
+
+    }
+
+    "not warn about HTTP server, client and pool specific parsing overrides" in {
+      // these are brought in through some trix in akka-http
+      val config = ConfigFactory
+        .parseString("""
+          akka.http.server.parsing.illegal-header-warnings = on
+          akka.http.client.parsing.illegal-header-warnings = on
+          akka.http.host-connection-pool.parsing.illegal-header-warnings = on
+        """)
+        .withFallback(reference)
+
+      val checker = new ConfigChecker(extSys, config, reference)
+      val warnings = checker.check().warnings
+
+      warnings should be(Vector.empty)
+    }
+
+    "not warn about things in the cinnamon config" in {
+      val config = ConfigFactory.parseResources("all-cinnamon-2018-05-14-reference.conf").withFallback(reference)
+
+      val checker = new ConfigChecker(extSys, config, reference)
+      val warnings = checker.check().warnings
+
+      warnings should be(Vector.empty)
+    }
+
+    @nowarn("msg=possible missing interpolator")
+    val defaultGrpc = ConfigFactory.parseString("""
+      akka.grpc.client."*" {
+        host = ""
+        port = 0
+        ssl-config = ${ssl-config}
+      }
+    """)
+
+    "not warn about paths with quotes" in {
+      val config = ConfigFactory
+        .parseString("""akka.grpc.client."*".ssl-config.hostnameVerifierClass = 37""")
+        .withFallback(reference)
+      val referenceWithGrpc = reference.withFallback(defaultGrpc).resolve
+      val checker = new ConfigChecker(extSys, config, referenceWithGrpc)
+      val warnings = checker.check().warnings
+
+      warnings should be(Vector.empty)
+    }
+
+    "warn about paths under akka.grpc.client for specific services" in {
+      val config = ConfigFactory
```

Comment:
This test is failing.
Here the problem is that is loading properties that still have `akka.remote.classic`.  I couldn't find the culprit today.

Reply:
found it